api/src/server/auth.rs

```diff
@@ -0,0 +1,144 @@
+use crate::{
+    error::ResponseError,
+    models::{NewUser, User},
+    schema::users,
+    util::Pool,
+};
+use actix_identity::Identity;
+use actix_web::{get, http, web, HttpResponse, Responder};
+use diesel::{prelude::RunQueryDsl, PgConnection};
+use openid::{Client, DiscoveredClient, Options, Token, Userinfo};
+use reqwest::Url;
+use serde::Deserialize;
+use std::{collections::HashMap, sync::RwLock};
+// TODO: make this a db table so its persitent throgh restarts
+pub struct Sessions {
+    pub map: HashMap<String, (User, Token, Userinfo)>,
+}
+/// Setup a client for a given provider
+/// based on the id, secret, and provider url
+pub async fn make_client(
+    client_id: String,
+    client_secret: String,
+    issuer_url: &str,
+) -> Client {
+    let redirect = Some(host("/api/oidc/callback"));
```

Comment:
Why does the client need to know the redirect in order to be created? Doesn't that only need to be known by the provider? Since the func takes an `Option`, what happens if you just pass `None`?

Reply:
Since you can specify multiple valid redirects in the provider you need to tell it which one to use when you initally send the user to the provider. Not sure why the lib made it an option type tbh